api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -136,6 +136,21 @@ public void saveEncounter_shouldUpdateExistingEncounterWhenAChildObsIsEdited() t
 	}
 
 	@Test
+	@Verifies(value = "should update leaf obs and not duplicate at encounter level", method = "saveEncounter(Encounter)")
+	public void saveEncounter_shouldUpdateValueOfLeafObsAndNotDuplicateAtEncounterLevel() throws Exception {
+		executeDataSet(ENC_OBS_HIERARCHY_DATA_XML);
+		Encounter encounter = Context.getEncounterService().getEncounter(100);
+		assertEquals(1, encounter.getObsAtTopLevel(true).size());
+		Obs topLevelObs = encounter.getObsAtTopLevel(true).iterator().next();
+		topLevelObs.getGroupMembers().iterator().next().setValueText("editing first obs");
+		encounter.addObs(topLevelObs);
+
+		Encounter savedEncounter = Context.getEncounterService().saveEncounter(encounter);
+
+		assertEquals(1, savedEncounter.getObsAtTopLevel(true).size());
```

Comment:
The thing is, if i dont flush and clear the session and try to resave the enconunter, the obs which is now attached directly to the encounter will get saved. So I thought I'll add a test around it. Without the line that I added, whenever the leaf obs is edited, a duplicate obs is created which is attached only to encounter, since we save encounter more than once in our use cases.